lib/puppet/info_service/class_information_service.rb

```diff
@@ -88,11 +88,10 @@ def extract_default(structure, p)
     structure
   end
 
-  def typeexpr_to_string(type_expr)
+  def typeexpr_to_string(name, type_expr)
     type_parser.interpret_any(type_expr, nil).to_s
-  rescue Puppet::ParseError
-    # type is to complex - contains expressions that are not literal
-    nil
+  rescue Puppet::ParseError => e
+    raise Puppet::Error, "The parameter '$#{name}' is invalid: #{e.message}", e.backtrace
```

Comment:
The important part is here. This raise is rescued by https://github.com/puppetlabs/puppet/blob/9b3514e6ad2ef37e24e551a055667ce882e6cf47/lib/puppet/info_service/class_information_service.rb#L65-L67 so that the API returns a meaning `error` hash.